flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ActiveResourceManagerFactory.java

```diff
@@ -81,4 +83,14 @@ private static Configuration createActiveResourceManagerConfiguration(Configurat
 		ClusterInformation clusterInformation,
 		@Nullable String webInterfaceUrl,
 		ResourceManagerMetricGroup resourceManagerMetricGroup) throws Exception;
+
+	protected WorkerResourceSpec createDefaultWorkerResourceSpec(Configuration configuration) {
```

Comment:
I tend to disagree, for the following reasons.
- The default `WorkerResourceSpec` is needed for both static and dynamic slot allocation. For dynamic allocation, we need it for scenarios where some jobs do not provide fine grained resource requirement.
- We need the default `WorkerResourceSpec` only for active RMs, but not for standalone RM. SM does not know which RM it runs on.
- We need the default CPU cores for creating the default `WorkerResourceSpec`, which is depending on different configuration options per active RM. SM does not have that information.
- According to the dependency injection principle, we should try to avoid creating dependencies inside the component if possible.